### PR TITLE
Provide parameters to retire and unretire a VM

### PR DESCRIPTION
**Pull request by TuxPowered42**

### CODE REVIEW

File: igvm/cli.py [hunk 1]
```diff
@@ -266,6 +266,11 @@ def parse_args():
         'vm_hostname',
         help='Hostname of the guest system',
     )
+    subparser.add_argument(
+        '--unretire',
+        action='store_true',
+        help='Unretire a VM, set it in maintenance mode',
+    )
```
Comment (kofrezo) [inline igvm/cli.py]: I think it would be easier to just have a target state switch and default one would be maintenance

Comment (TuxPowered42) [reply igvm/cli.py]: True, I'll implement it.

File: igvm/commands.py [hunk 2]
```diff
@@ -315,6 +323,11 @@ def vm_stop(vm_hostname, force=False):
         else:
             vm.shutdown()
         log.info('"{}" is stopped.'.format(vm.fqdn))
+        if retire:
+            vm.dataset_obj['state'] = 'retired'
+            vm.dataset_obj.commit()
+            log.info('"{}" is retired.'.format(vm.fqdn))
+
```
Comment (HerrSpace) [inline igvm/commands.py]: Setting the state before shutdown would make testtool take the VM out of potential LBs before it goes down. Can we move this?

File: igvm/commands.py [hunk 3]
```diff
@@ -202,7 +202,7 @@ def vm_migrate(vm_hostname, hypervisor_hostname=None, newip=None,
         vm = es.enter_context(
             _get_vm(
                 vm_hostname, ignore_reserved=ignore_reserved,
-                ignore_retired=True,
+                allow_retired=True,
```
Comment (HerrSpace) [inline igvm/commands.py]: `ignore_reserved` is talking about the hv and `allow_retired` about the vm. Maybe we can call them `ignore_reserved_

I propose renaming `ignore_reserved` into `allow_reserved_hv` and `allow_retired` into `allow_retired_vm`.

Comment (TuxPowered42) [reply igvm/commands.py]: I think there is something very wrong with the code. `ignore_reserved` passed to `_get_vm()` has no effect on anything at all. I will just rip it out.

File: igvm/commands.py (hunk 2 above)
Comment (TuxPowered42) [reply igvm/commands.py]: Sure, I can do it.

**Comment by TuxPowered42** [comment]
retest this please

**Comment by TuxPowered42** [comment]
retest this please

File: igvm/commands.py (hunk 3 above)
Comment (TuxPowered42) [reply igvm/commands.py]: Ripped out.

File: igvm/commands.py (hunk 2 above)
Comment (TuxPowered42) [reply igvm/commands.py]: It is done.

File: igvm/cli.py (hunk 1 above)
Comment (TuxPowered42) [reply igvm/cli.py]: It is implemented.

**Pull request merged**